wallet/src/main/java/com/android/identity_credential/wallet/ui/TimeAgo.kt

```diff
@@ -0,0 +1,116 @@
+package com.android.identity_credential.wallet.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import com.android.identity_credential.wallet.R
+import kotlinx.coroutines.delay
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimePeriod
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.daysUntil
+import kotlinx.datetime.minus
+import kotlinx.datetime.monthsUntil
+import kotlinx.datetime.yearsUntil
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Format time interval to the time instant in the past in human-readable way, such as
+ * "3 days and 5 hours ago".
+ *
+ * [instant] time instant in the past
+ * [now] time now, only exposed for testing
+ */
+@Composable
+fun TimeAgo(instant: Instant, now: Instant = Clock.System.now()): String {
```

Comment:
I will beef it up for the future times. In (unsurprisingly) English and (somewhat surprisingly) Russian it is enough to format the duration and then combine it with the words that indicate past or future. (i.e. in Russian "5 hours ago"/"5 hours from now" will be "5 часов назад"/"через 5 часов") - I don't know if that's just luck or there is a general rule. I will not do separate past/future localization for now, just add a TODO to figure it out.

"two hours" instead of "2 hours" is a bit of extra work for English, but it not generally translatable, at least once you have grammatical gender. In Russian "two seconds" is "две секунды" and "two hours" is "два часа". I think ICU plurals library handles these sort of stuff better if we really want to do it.

Reply:
Future instants and week units are now supported (but for durations over a month, using months and days, not weeks).